server_test.go

```diff
@@ -2,13 +2,204 @@ package veneur
 
 import (
 	"bytes"
+	"compress/zlib"
+	"encoding/json"
 	"math/rand"
+	"net/http"
+	"net/http/httptest"
+	"sync"
 	"testing"
 	"time"
 
 	"github.com/stretchr/testify/assert"
 )
 
+// set up a boilerplate local config for later use
+func localConfig() Config {
+	return Config{
+		APIHostname:         "http://localhost",
+		Debug:               false,
+		Hostname:            "localhost",
+		Interval:            100 * time.Millisecond,
+		Key:                 "",
+		MetricMaxLength:     4096,
+		Percentiles:         []float64{.5, .75, .99},
+		ReadBufferSizeBytes: 2097152,
+		UDPAddr:             "localhost:8126",
+		HTTPAddr:            "localhost:8127",
+		ForwardAddr:         "http://localhost",
+		NumWorkers:          96,
+		NumReaders:          4,
+		StatsAddr:           "localhost:8125",
+		Tags:                []string{},
+		SentryDSN:           "",
+		FlushLimit:          1024,
+	}
+}
+
+// assertMetrics checks that all expected metrics are present
+// and have the correct value
+func assertMetrics(t *testing.T, metrics DDMetricsRequest, expectedMetrics map[string]float64) {
+	// it doesn't count as accidentally quadratic if it's intentional
+	for metricName, expectedValue := range expectedMetrics {
+		assertMetric(t, metrics, metricName, expectedValue)
+	}
+}
+
+func assertMetric(t *testing.T, metrics DDMetricsRequest, metricName string, value float64) {
+	defer func() {
+		if r := recover(); r != nil {
+			assert.Fail(t, "error extracting metrics", r)
+		}
+	}()
+	for _, metric := range metrics.Series {
+		if metric.Name == metricName {
+			assert.Equal(t, int(value+.5), int(metric.Value[0][1]+.5), metricName)
+			return
+		}
+	}
+	assert.Fail(t, "did not find expected metric", metricName)
+}
+
+// setupLocalServer creates a local server from the specified config
+// and starts listening for requests. It returns the server for inspection.
+func setupLocalServer(t *testing.T, config Config) Server {
```

Comment:
> (if there's nothing to flush, the API server is never contacted at all)

oh right - good catch

related to that, we might want to add a comment for statsaddr in the config, to clarify that we're intentionally black-holing internal metrics (since otherwise those could slip into the flushes under test and muddle up the assertion)